src/spike/spike.service.ts

```diff
@@ -1,116 +1,43 @@
-import * as https from 'https';
-import { RedisClient } from 'redis';
-import { promisify } from 'util';
-import SpikeToken from './spike.token.interface';
-import { tokenModel, IToken, Second } from './spike.model';
-import { log, Severity } from '../logger';
-import { tokenID } from '../config';
-import { ServerError } from '../utils/errors';
+import { SpikeError } from '../utils/errors';
+
+const protoLoader = require('@grpc/proto-loader');
+const grpc = require('grpc');
+
+const PROTO_PATH =  `${__dirname}/../../proto/spike.proto`;
+const packageDefinition = protoLoader.loadSync(
+    PROTO_PATH,
+    {
+        keepCase: true,
+        longs: String,
+        enums: String,
+        defaults: true,
+        oneofs: true,
+    });
+const spike_proto = grpc.loadPackageDefinition(packageDefinition).spike;
 
 export default class Spike {
-    private redis: RedisClient;
-    private token: IToken | null | void;
-    private spikeId: string;
-    private spikeSecret: string;
-    private spikeURL: string;
-    private getAsyncRedis: any;
 
-    /**
-     * @param redisInstance - Redis instance to get\set token from redis service
-     */
-    constructor(redis: RedisClient) {
-        this.redis = redis;
-        this.token = undefined;
-        this.getAsyncRedis = promisify(this.redis.get).bind(this.redis);
-        this.spikeId =  process.env.SPIKE_CLIENT_ID || '';
-        this.spikeSecret = process.env.SPIKE_CLIENT_SECRET || '';
-        this.spikeURL = process.env.SPIKE_TOKEN_URL || '';
-    }
-
-    /**
-     * Renews the kartoffel token from spike
-     */
-    private async renewToken(): Promise<SpikeToken> {
-        // For when the Spike's https certificate is self signed.
-        const authorizationValue:string = Buffer.from(`${this.spikeId}:${this.spikeSecret}`).toString('base64');
-
-        const res: AxiosResponse = await axios({
-            httpsAgent: new https.Agent({ rejectUnauthorized: false }),
-            method: 'post',
-            url: this.spikeURL,
-            data: {
-                grant_type: 'client_credentials',
-                audience: 'kartoffel',
-            },
-            headers: { Authorization: `Basic ${authorizationValue}` },
-        });
-        return res.data;
-    }
-
-    /**
-     * Returns a kartoffel token saved in redis or from spike
-     */
-    public async getToken (): Promise<string> {
-        if (this.token && this.checkTokenValidity()) {
-            return this.token.token;
-        }
-        log(Severity.INFO, 'attempting to obtain token from redis', 'getToken');
-        let kartoffelToken:string|null = await this.getAsyncRedis('kartoffel:token');
-        let updateMongo: boolean = false;
-        let tokenExp: number = 0;
-        if (kartoffelToken) {
-            // TODO: Token verification?
-            log(Severity.INFO, 'successfully obtained token', 'getToken');
-            return kartoffelToken;
-        }
+    public async getToken() {
```

Comment:
Add an explanation about the class and about the methods inside.
Add the return type in the method statement. should something like 'Promise<THE_TYPE>'